### PR TITLE
Migrate to Statie

### LINKED ISSUE #1: Migrate to Statie?
Hi Tomas, 

I've noticed you have 2 repositories for 1 websites - This and https://github.com/tomasfejfar/tomasfejfar.cz

Not sure what your plans in the future are, but converting this page from Jekyll to Statie would a huge challange.

I'm just writing a [post about Jekyll → Statie migration](https://github.com/TomasVotruba/tomasvotruba.cz/pull/631) and this would be huge proof of concept :)

Would you care for such pull-request from me?

### PR DESCRIPTION
**Pull request by TomasVotruba**
Closes #1 

I copied most of the design from  https://www.tomasfejfar.cz as we discussed.

I usually comment what changed, why and how Statie works, but as you already use it, I would explain what you already know. If you have some question, just ask anything. I'll answer you.

<br>

Next, after this merge, I think the best would be to send this website as pull-request to https://github.com/tomasfejfar/tomasfejfar.cz. Where blog would be in new page - http://tomasfejfar.cz/blog 

There you could polish the design, now it would be doing duplicated work.

<br>

Then, you could redirect the subdomain and remove this repository. If you need help with Disquss migration, I can give you a hand.

What do you think? :)

### Homepage

image

### All Posts

image

### Post Detail

image

### CODE REVIEW

**Comment by TomasVotruba** [comment]
Ready to merge

**Comment by tomasfejfar** [comment]
Thank you, I'll check it out!

**Comment by tomasfejfar** [comment]
For some reason `exec('php -S localhost:8000 -t output');` in gulp does not work in docker (https://github.com/Symplify/Symplify/pull/1363).

**Comment by tomasfejfar** [comment]
Could you allow me to push to your branch? It's called something like "allow admin to change this PR" or something like that... I want to get everything ready before I merge it a break my current blog.

**Comment by TomasVotruba** [comment]
I'll try to find it. I thought it's by default

**Comment by TomasVotruba** [comment]
"Allow edits from maintainers" - it was on, I rechecked it. Should work since start

**Comment by tomasfejfar** [comment]
Weird... 

```
"C:\Program Files\Git\bin\git.exe" push -u --recurse-submodules=check --progress "git://github.com/TomasVotruba/blog-1.git" refs/heads/review/2:refs/heads/migrate-to-statie-next
fatal: remote error:
  You can't push to git://github.com/TomasVotruba/blog-1.git
  Use https://github.com/TomasVotruba/blog-1.git
Done
```

**Comment by TomasVotruba** [comment]
I merge these locally in new PR. This Github feature didn't work well for me either

Does `https://github.com/TomasVotruba/blog-1.git` help?

**Comment by tomasfejfar** [comment]
Huh, that worked, but forced me to login using my credentials instead of SSH key..

**Comment by TomasVotruba** [comment]
Yea, that's stupid feature. I saw other people merge it locally as well.

Also, I don't expect anyone to add anyone commits to my PR. Once coleague added few commits and I didn't know and when force-pushed fixup commits, his work went away.

File: statie.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ parameters:
         avatar: 'bio-photo.jpg'
         bio: 'Web developer'
         email: 'tomas.fejfar@gmail.com'
-        disqus-shortname: 'tomasfejfar'
+        disqusShortname: 'tomasfejfar'
```
Comment (TomasVotruba) [inline statie.yaml]: `-` is ignored somehow or what happened? 

I created a command to handle migration, so it should cover this.

File: gulpfile.js [hunk 2]
```diff
@@ -18,6 +18,7 @@ gulp.task('default', function () {
 
     // For the second arg see: https://github.com/floatdrop/gulp-watch/issues/242#issuecomment-230209702
     return watch(['source/**/*', '!**/*___jb_tmp___'], function () {
+        console.log(' [PROCESSING] Regenerating');
```
Comment (TomasVotruba) [inline gulpfile.js]: Good idea actually! Could you add that to Statie `gulpfile.js`?

**Comment by tomasfejfar** [comment]
Since I've been using `git push --force-with-lease` I've never overwritten anyone's work. Do you use it? 

From the commits I think 1a4e364 is significant for your efforts of scripting the migration. 

It seems like a regex or something is off somewhere is a variable contains dash (`some-variable`). The include was malformed and the variable did not work. 

I imagine it would be hard to detect and fix (you'd need to parse the templates IMHO). So at least a warning that config contains dashed variable might be raised.

File: gulpfile.js (hunk 2 above)
Comment (tomasfejfar) [reply gulpfile.js]: Yes, I'll send a PR...

Comment (tomasfejfar) [reply gulpfile.js]: There is one further issue though. The exec is asynchronous, so sometimes the processings and the message from console get mixed up :( 

I'm not sure, but I think using [`execSync`](https://nodejs.org/api/child_process.html#child_process_child_process_execsync_command_options) migth fix that.

Comment (TomasVotruba) [reply gulpfile.js]: I see. If you find a good enough way that makes sense, please share it. Otherwise no need to waste time on this.

**Comment by TomasVotruba** [comment]
> Since I've been using git push --force-with-lease I've never overwritten anyone's work. Do you use it?

I think I tried it but it got some extra maintenance so I dropped it. Will try it for alias now to verify.

> From the commits I think 1a4e364 is significant for your efforts of scripting the migration.

Yea, I commented there. Not sure if yaml/twig can read it, since I use `_` everywhere.

> The include was malformed and the variable did not work.

There include needs brackets. Already got that covered in command.

> I imagine it would be hard to detect and fix (you'd need to parse the templates IMHO). So at least a warning that config contains dashed variable might be raised.

Not really, just find params with `-`, rename it to `_` and do so in templates.

**Comment by TomasVotruba** [comment]
Made an issue: https://github.com/Symplify/Symplify/issues/1364 I hope I got the problem right

File: gulpfile.js (hunk 2 above)
Comment (tomasfejfar) [reply gulpfile.js]: https://github.com/Symplify/Symplify/pull/1365

execSync does not work as expected...

**Pull request closed without merging**